src/net/codec.rs

```diff
@@ -761,6 +766,42 @@ impl StacksMessageCodec for NatPunchData {
     }
 }
 
+impl StacksMessageCodec for MemPoolSyncData {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
+        match *self {
+            MemPoolSyncData::BloomFilter(ref bloom_filter) => {
+                write_next(fd, &(MemPoolSyncDataID::BloomFilter as u8))?;
+                write_next(fd, bloom_filter)?;
+            }
+            MemPoolSyncData::TxTags(ref seed, ref tags) => {
+                write_next(fd, &(MemPoolSyncDataID::TxTags as u8))?;
+                write_next(fd, seed)?;
+                write_next(fd, tags)?;
+            }
+        }
+        Ok(())
+    }
+
+    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<MemPoolSyncData, codec_error> {
+        let data_id: u8 = read_next(fd)?;
+        match data_id {
+            x if x == MemPoolSyncDataID::BloomFilter as u8 => {
+                let bloom_filter: BloomFilter<BloomNodeHasher> = read_next(fd)?;
+                Ok(MemPoolSyncData::BloomFilter(bloom_filter))
+            }
+            x if x == MemPoolSyncDataID::TxTags as u8 => {
+                let seed: [u8; 32] = read_next(fd)?;
+                let txtags: Vec<TxTag> = read_next(fd)?;
+                Ok(MemPoolSyncData::TxTags(seed, txtags))
+            }
+            _ => Err(codec_error::DeserializeError(format!(
+                "Unrecognized MemPoolSyncDataID {}",
+                &data_id
+            ))),
```

Comment:
I dropped a note at the definition for MemPoolSyncDataID, but this match should take advantage of the `from_u8` method from the `define_u8_enum` variant. That way, we don't need to do the `x if x == .. as u8` matching pattern (which circumvent's rust's variant coverage checks).

Reply:
Yup, done